auto_create.go

```diff
@@ -78,6 +78,9 @@ type autoCreateResourceInfo struct {
 	autoCreateLabelValues []string
 }
 
+/* map to track app name being added or deleted for an auto creation name */
+var autoCreateAppNameMap = make(map[string][]string)  
```

Comment:
This map is application[]component[].  It needs to also include namespace, since auto-create is a namespace-scoped operation.  The result is that components from other namespaces are counted as components of a given application of the same name when they are not.  This prevents an application in namespace X from being deleted even when all its auto-create components in namespace X have been deleted.